gromacs/gromacs-2021.x.patch

```diff
@@ -502,30 +505,5 @@ index c2973bb1af..cb4d1da254 100644
 +           Generic data file
 + -colvars_restart [&lt;.dat&gt;]  (colvars.dat)    (Opt.)
 +           Generic data file
- 
+
  Options to specify output files:
- 
-diff --git a/.github/workflows/build_cmake.yml b/.github/workflows/build_cmake.yml
```

Comment:
So this part of the patch isn't needed anymore? It's better this way, we don't want to be altering these build recipes.

Reply:
Well this part should not be in the patch file since it concerns only the Colvars repo and not the GROMACS source tree

Reply:
@HubLot This patch that you are removing was introduced in bbe94155 and meant to a CI error in standard GROMACS 2021: 
https://github.com/gromacs/gromacs/actions/runs/3629868210/jobs/6122561771
This error was due to a change in the macOS Azure instances available, and the GROMACS people fixed it in 2022 and later. But 2021 seems to be already lagging behind even though 2023 is still in preview...

Ideally, they should backport the CI fix in 2021.  But if we didn't do it in the patch, the CI jobs on the 2021 branch of our own fork  for reasons that are unrelated to Colvars. Whether or not you are in the mood to check the details (I probably wouldn't) it's impossible to see them without being logged into GitHub.

Reply:
Thanks for the explanation.
I understand the issue but this patch should not be in this file. This gromacs-2021.x.patch is meant to be used with a standard GROMACS source tree on a standard user computer. 
Right now, the `update-colvars-code.sh` is "failing" because `patch` binary doesn't know what do to with those lines (path *.github/workflow/...* is unknown).

So, it should remain in CI somehow

Reply:
Ah, ok. In my view, the best-supported would be to download one of the pre-patched releases, rather than downloading a standard release and patching it ourselves.

That said, it's safer to add a check for this, which I did in `master`, more precisely f3808039. Can you rebase this branch over that?

Reply:
Thanks Giacomo, this works, although it adds a few very specialized lines to user-facing `update-colvars-code.sh` which is getting a little heavy. I wonder if that would be handled directly within the CI scripts ?

Reply:
Feel free to move them, but only in a way such that those lines are run regardless of whether you are on a user desktop or an Actions instance.

Reply:
Fair enough. Since I don't see an simple way to do this, I'm ok with keeping it the way it is in current master.

Reply:
okay for the fix in the update-colvars.code.sh 
I'll rebase the branch